### ISSUE TITLE
Allow to specify `AbortSignal` in `startRegistration` and `startAuthentication`

### ISSUE THREAD

**Issue by mahnunchik** [MasterKale/SimpleWebAuthn#448]
## Describe the issue

There is no way to stop registration/authentication process. This is necessary when the user canceled an action in the UI, for example, pressed the back button.

## Expected behavior

```js
const controller = new AbortController();
startAuthentication({ signal: controller.signal});

// Cancel action
controller.abort();
```

## Dependencies

- **OS:** all
- **Browser:** all
- **Authenticator:** all

### SimpleWebAuthn Libraries

Latest version

## Additional context

`AbortController` is used inside library but there is no option to specify it outside.

**Comment by MasterKale** [comment]
SimpleWebAuthn manages its own use of AbortController, what is an example of a scenario in which this is insufficient? You mention the user hitting the back button, are we talking a basic server-rendered website? A single-page app? If the browser doesn't cancel the WebAuthn call on a navigation event then I'm not sure how specifying a custom instance of an abort controller would be helpful here.

**Comment by mahnunchik** [comment]
I'm working on single-page app (web, electron). User can leave registration/authentication view (cancel action). Of course, after the timeout registration/authentication process will be cancelled, but it would be useful to be able to cancel it from the app.

**Comment by MasterKale** [comment]
I've created #449 that will offer a new `WebAuthnAbortService` import with a `cancelCeremony()` method that can be called to cancel any in-flight WebAuthn ceremony started with `startRegistration()` or `startAuthentication()`.

**Comment by mahnunchik** [comment]
This is exactly what I need for the current project.

**Comment by MasterKale** [comment]
`WebAuthnAbortService.cancelCeremony()` is now available as of **@simplewebauthn/browser@8.3.0** ✌️

**Comment by martpet** [comment]
I would like to supply my own `AbortSignal` to `startRegistration()` as I want to reuse it in `fetch` calls to /options and /verify.
